runtime/runtime-params-estimator/estimator-warehouse/src/import.rs

```diff
@@ -173,11 +165,34 @@ mod test {
     }
 
     impl Db {
+        /// Create a new in-memory test database.
         pub(crate) fn test() -> Self {
             let conn = Connection::open_in_memory().unwrap();
             let init_sql = include_str!("init.sql");
-            conn.execute(init_sql, []).unwrap();
+            conn.execute_batch(init_sql).unwrap();
             Self::new(conn)
         }
+
+        /// Create a new in-memory test database with data defined by the input.
+        ///
+        /// The test data is expected to come in blocks of JSON lines with a commit header.
+        /// WAIT statements can be used as barriers to make sure DB timestamps differ.
+        pub(crate) fn test_with_data(input: &str) -> Self {
+            let db = Self::test();
+            for block in input.split("\n\n") {
+                if block.trim() == "WAIT" {
+                    // Wait 1s to ensure the following data is considered to be later
+                    std::thread::sleep(std::time::Duration::new(1, 0));
```

Comment:
* minor comment: `Duration::from_secs(1)`
* major comment -- sleeping in tests seems quite suboptimal -- sleeps is how test suites become unreasonably slow (well, actually there's bimodal distribution of slow test-suites -- either the tests are slow because CPU does nothing waiting for sleep (common case); or the tests do some kind of exhaustive search over all execution paths (good case you want to be in)). A better way to deal with such situations is by inserting time from outside. I *think* for sqlite you want 
  * https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.create_scalar_function
  * https://stackoverflow.com/a/71307756

  As a side benefit, this'll get us rid of the redactions and associated regex dependency. Redactions are a telling smell -- better to avoid them.

Reply:
It didn't occur to me that I could simply mock the datetime function. Now that makes so much more sense than what I was doing.

Reply:
>It didn't occur to me

My exact train of thought here was:

1. wow, there's regex in Cargo.toml, that's a) bad b/c its slow to compile and b) suspect as why would we need regex? 
2. so, we use regex for redactions. That's a smell -- better to control env to not need redactions.
3. so, we redact-out the time -- should be easy fix, just pass `Instant` as field in the `ImportConfig`
4. oh, so we even actually sleep in the tests? That's such a relief, now I can make a strong case in the review instead of vague complains about regex  :-)
5. huh, so it's pretty curious, the time comes from SQLite, so we can't *just* pass the instant. SQLite is a solid piece of tech, so it *must* have ability to substitute externals
6. googles "SQLite mock time"
7. Ok, so SQLite allows to just override any built-in function? That's nifty design!
8. How come there's so little info on mocking time in SQLite? It's impossible that people don't have this problem, so everyone must be sleeping I guess :(